### PR TITLE
Propagate SQL exception from queryTopology

### PR DESCRIPTION
**Pull request by karenc-bq**
### Summary

Propagate SQL exception from queryTopology

### Description

Propagate SQL exception from queryTopology so it can be handled by caller instead of ignoring the exception

### Additional Reviewers

@sergiyv-bitquill 
@ColinKYuen

### CODE REVIEW

File: src/main/user-impl/java/com/mysql/cj/jdbc/ha/plugins/failover/ClusterAwareWriterFailoverHandler.java [hunk 1]
```diff
@@ -298,7 +298,7 @@ public WaitForNewWriterHandler(
       this.originalWriterHost = currentHost;
     }
 
-    public WriterFailoverResult call() {
+    public WriterFailoverResult call() throws SQLException {
```
Comment (sergiyv-bitquill) [inline src/main/user-impl/java/com/mysql/cj/jdbc/ha/plugins/failover/ClusterAwareWriterFailoverHandler.java:301]: I'd prefer to catch and handle exceptions inside this task. Bubbling up exception may have side effects on entire failover process.

File: src/main/user-impl/java/com/mysql/cj/jdbc/ha/plugins/failover/ClusterAwareWriterFailoverHandler.java [hunk 2]
```diff
@@ -368,7 +368,8 @@ private boolean isValidReaderConnection(ReaderFailoverResult result) {
      *
      * @return Returns true if successful.
      */
-    private boolean refreshTopologyAndConnectToNewWriter() throws InterruptedException {
+    private boolean refreshTopologyAndConnectToNewWriter()
```
Comment (sergiyv-bitquill) [inline src/main/user-impl/java/com/mysql/cj/jdbc/ha/plugins/failover/ClusterAwareWriterFailoverHandler.java:371]: Similar note. We need to catch and handle exceptions inside this method.

File: src/main/user-impl/java/com/mysql/cj/jdbc/ha/plugins/failover/FailoverConnectionPlugin.java [hunk 3]
```diff
@@ -234,9 +234,9 @@ public Object execute(
 
     Object result = null;
 
-    updateTopologyAndConnectIfNeeded(false);
 
     try {
+      updateTopologyAndConnectIfNeeded(false);
```
Comment (sergiyv-bitquill) [inline src/main/user-impl/java/com/mysql/cj/jdbc/ha/plugins/failover/FailoverConnectionPlugin.java:239]: We need to remove a logic that handles closed connections inside this method (line 631)

Comment (karenc-bq) [reply src/main/user-impl/java/com/mysql/cj/jdbc/ha/plugins/failover/FailoverConnectionPlugin.java:239]: If the connection is closed when we call getTopology an exception will be thrown so we will never reach 631.
I am worried about the case where connection is closed but no exception is raised.

**Review by sergiyv-bitquill — APPROVED** [review]

**Pull request merged**